### DISCUSSION TITLE
Welcome to ChroniquesContemporaines Discussions!

### DISCUSSION THREAD

**Discussion by stephaned68** [stephaned68/ChroniquesContemporaines#9] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by romdesgrouas** [comment]
Bonjour,

Serait-il possible d'utiliser les modificateurs de test au lieu des modificateurs de base pour les DM ?
Par exemple, si mon personnage à un buff de -8 en force, celui-ci est bien pris en compte pour mon jet d'attaque, mais pas pour mon jet de DM qui dépend aussi de la force.

Merci d'avance

**Reply by stephaned68** [reply]
Bonjour

C'est curieux, le mod qui entre dans le calcul de l'attaque au CAC est le mod de base de FOR, pas le mod de test
Donc un buff de -8 à la FOR ne devrait pas se reporter sur l'attaque au CAC ?

En fait dans les règles de CO, le mod de FOR entre dans le calcul du score d'attaque au CAC et est ajouté aux DM des armes de CAC, mais si la FOR subit un buff, ni le score d'attaque ni les DM au CAC ne sont sensés être modifiés. Seuls les jets de FOR sont concernés par le buff, pas les scores dérivés

**Reply by romdesgrouas** [reply]
Bonjour,

Pardon, j'ai été un peu trop vite dans mon explication ....
Dans mon buff je mets : "FOR : -8 ; ATC : -8", d'où me -8 sur l'attaque au CAC qui n'est en effet pas dû au buff de force.

Merci pour l'explication, j'utiliserais la partie "Modificateurs de Dommages" du coup, il faut juste que je pense à le décocher pour les attaques à distance.

**Reply by stephaned68** [reply]
Je dois pouvoir ajouter les mod de test à la liste des mod de caractéristiques pour les DM sans trop de difficulté. 
Je regarde ça le week-end prochain

**Reply by romdesgrouas** [reply]
Ce serait super, merci !

**Reply by stephaned68** [reply]
@romdesgrouas 
Je viens de pousser la version 3.11.0 de la fiche COC avec deux nouveautés : 
- possibilité d'utiliser les mods de test comme bonus aux DM
- possibilité d'afficher le token du personnage dans les jets de dés du chat

Si tu as un compte pro et une partie avec fiche "custom", tu peux d'ores et déjà récupérer le fichier coc.html et l'insérer dans les paramètres de ta partie. Sinon il faudra attendre entre mardi et jeudi pour que la mise à jour que je vais pousser chez Roll20 soit diffusée dans les parties existantes.

**Reply by romdesgrouas** [reply]
Super, merci beaucoup ça fonctionne très bien !

**Reply by stephaned68** [reply]
Cool !
N'hésite pas à utiliser l'onglet Issues si tu trouves des bugs ou si tu as d'autre demandes d'amélioration, j'ai créé des templates de tickets pour les deux types de demandes
Bien à toi 
Stéphane

**Comment by Xurantar** [comment]
Salut Stephaned, 

J'ai vus sur le fils de discussion de Black Book Edition ( https://black-book-editions.fr/forums.php?topic_id=11691&nbp=8&nop=5 ) que sur la fiche Roll20 de Chronique oublié contemporain, tu envisageait de mettre le compteur en liste. (Il y a un petit moment, environs 2 ans)

Est-ce que tu pense que c'est une modification réalisable ? 
Ou augmenté le nombre de compteur de la fiche à 2 ou 3 ? 

Je suis confronté à des situations où via les différentes voies offertes par le livre de base, mes joueurs se retrouve avec plusieurs ressources à gérées, mais il n'y a qu'un seul compteur ! 

(Exemple : La voie des sciences, la voie des art, donnant tout deux, une ressource supplémentaire sous forme de Point Tactique, et Point d'inspiration, qui sont employé plus ou moins rapidement au cours des combats !)

Il est possible de noter ça dans "autre ressources", dans l'onglet d'équipement de la fiche, bien sur, mais l'avantage des compteurs est qu'il sont sur le même onglet que les jets de caractéristiques et d'attaque, et que l'on peut faire appel à eux dans des macros ! 

Je te remercie pour le travail que tu à déjà effectué ! Cette fiche est superbe, et notre outils de base, sur bien des campagnes ! 
Merci !

**Reply by stephaned68** [reply]
Salut Xurantar

Les trois dernières années ont été un peu folklo question taf et je n'ai plus trop touché à la fiche COC. J'ai repris récemment la fiche COG pour la sortie de la V2 dont le PDF a été fourni aux souscripteurs de la PP.

Je vais voir pour ajouter une liste de compteurs, ça ne devrait pas être trop compliqué.

**Reply by Xurantar** [reply]
Merci beaucoup !

**Reply by stephaned68** [reply]
Re

Si tu veux utiliser les compteurs dans des macros, une liste ne va pas être pratique. Les noms des attributs sont en effet du type @{repeating_nomsection_$N_nomattribut} ou N va de 0 au nombre de lignes -1. Peut-être que rajouter 2 compteurs serait plus pratique ?
Dis-moi ce que tu préfères :)

**Reply by Xurantar** [reply]
Rajouter deux compteurs serait plus pratique donc je pense ! Ca semble être la bonne solution.

**Reply by stephaned68** [reply]
J'ai fait ce matin les modifications nécessaires pour ajouter les compteurs 2 & 3. J'en ai profité pour ajouter une fonctionnalité pour faciliter l'utilisation des compteurs dans les macros.
Par exemple, si on indique en compteur 1 'Points d'Inspiration' avec en abréviation PI, dès que la valeur change, ça crée ou met à jour un attribut nommé 'cptr_pi' ce qui permet d'utiliser la syntaxe @{cptr_pi} dans les macros, ce qui les rend plus claires.

**Reply by stephaned68** [reply]
Je viens de pousser la version 3.14.0 de la fiche chez Roll20, ça devrait être disponible sous 2-3 jours

**Reply by stephaned68** [reply]
La version 3.14.0 vient d'être intégrée dans Roll20. Enjoy 😀

**Reply by Xurantar** [reply]
Merci beaucoup !

**Comment by romdesgrouas** [comment]
Bonjour,

en explorant la fiche de perso pour faire évoluer mon univers, je me suis rendu compte que l'on pouvait personnaliser le nom  pour les Points de Mana
<img width="197" height="68" alt="image" src="https://github.com/user-attachments/assets/c8807bcc-8e50-4063-b200-2d6a4fe0846b" />
mais pas pour l'Attaque Magique
Est-ce que ce serait compliqué à ajouter ?

**Reply by stephaned68** [reply]
Bonjour

Oula, il a fallu que je me replonge dans ce code que je n'ai pas ouvert depuis deux ans au moins pour retrouver de quoi il retourne
C'est pour l'univers Menace-X que j'avais utilisé les attributs PM et PM_max en tant que points d'énergie PSI. On change l'intitulé quand cet univers est sélectionné dans la config de la fiche. C'est en dur dans le code javascript et pas modifiable "facilement" par l'utilisateur, à part dans l'onglet "Attributes & Abilities" de la fiche où tu les as trouvé.

Je n'ai pas vraiment l'intention de continuer à faire évoluer la fiche COC (trop de foutage de gueule de BBe sur le digital). 
Ce que je peux faire, c'est remplacer les textes en dur par des références à des attributs (ex: `ATKMAG_desc`), et tu irais le changer à ta convenance dans l'onglet "Attributes & Abilities" de la fiche.
Je regarde ça à l'occasion

**Reply by stephaned68** [reply]
J'ai commencé à regarder. Pas très compliqué :
<img width="381" height="242" alt="image" src="https://github.com/user-attachments/assets/e4419545-f13f-4a8c-98fd-6cbde47e7409" />
<img width="357" height="138" alt="image" src="https://github.com/user-attachments/assets/48b7cc96-a512-4d4f-8bf2-06964077f6c2" />
<img width="302" height="127" alt="image" src="https://github.com/user-attachments/assets/09cd9df1-a9b7-43b3-a8ce-e3a88506f78d" />
Tant que j'y suis, y a-t-il d'autres champs dont tu souhaiterais que les labels soient paramétrables ?

**Reply by romdesgrouas** [reply]
Bonjour,

Merci beaucoup pour la réactivité, !
je jette un oeil cet aprem et je reviens vers toi

**Reply by stephaned68** [reply]
Attention, j'ai expérimenté la modif sur ma campagne sandbox, mais ce n'est pas encore livré chez Roll20 (à partir du moment où c'est poussé chez eux, il faut quelques jours avant que ce soit mis à disposition sur la plate-forme)
C'est simplement que quitte à s'embarquer dans une pull-request chez R20, autant qu'il y ait tout ce qui peut t'être utile dedans

**Reply by romdesgrouas** [reply]
J'ai regardé, je ne vois rien d'autre, merci beaucoup

**Reply by stephaned68** [reply]
La version 3.21.0 a été envoyée à Roll20. 
Je te préviens quand je reçois la notification de merge sur la plate-forme

**Reply by stephaned68** [reply]
La PR est toujours en attente d'intégration depuis 6 jours. On va dire que c'est à cause des vacances...

**Reply by stephaned68** [reply]
La nouvelle version a été intégrée à Roll20
Ça devrait être disponible sur la plateforme incessamment sous peu

**Reply by romdesgrouas** [reply]
Merci pour le suivi de la maj sur roll20, c'est vraiment super sympa !
Je viens de me connecter et je vois bien la maj dans la suivi de version de la fiche
<img width="942" height="146" alt="image" src="https://github.com/user-attachments/assets/8410d671-6b1a-412d-9d4b-3118f8e6a056" />
mais je ne trouve pas "ATKMAG_desc" dans l'onglet "Attributs & capacités"
j'ai fait une recherche avec "mag" pour voir sir des fois c'était une autre "trad" une fois la PR intégrée sur roll20, mais je n'ai rien trouvé

**Reply by stephaned68** [reply]
J'ai vérifié, la version intégrée comprend bien la modification. Quand j'ai testé dans le sandbox, le VTT avait créé l'attribut il me semble. As-tu essayé de simplement créer l'attribut à la main et de mettre ton libellé dans la première case ?
